pkg/components/modelserver/modelserver.go

```diff
@@ -216,35 +219,47 @@ func NewConfigMap(d components.Detail, k *v1alpha1.Kepler) *corev1.ConfigMap {
 			Namespace: components.Namespace,
 			Labels:    labels,
 		},
-		Data: msConfig.Merge(trainerSettings).ToMap(),
+		Data: msConfig.ToMap(),
 	}
 }
 
 func settingsForTrainer(trainer *v1alpha1.ModelServerTrainerSpec) k8s.StringMap {
-	if trainer == nil {
-		return nil
-	}
+	msConfig := k8s.StringMap{}
+
+	prom := trainer.Prom
+	if prom != nil {
+		promHeaders := strings.Builder{}
+		interval := defaultPromInterval
+		step := defaultPromStep
+		// iterate through available headers and append to promHeaders (this string will be converted to dict in the image)
+		for _, h := range prom.Headers {
+			promHeaders.WriteString(h.Key)
+			promHeaders.WriteString(":")
+			promHeaders.WriteString(h.Value)
+			promHeaders.WriteString(",")
```

Comment:
is a trailing comma ok?

Reply:
I'm not sure who wrote this part. It will be used here: https://github.com/sustainable-computing-io/kepler-model-server/blob/4af262f2bb2ae2420eaddf64cdc8480eb05ff3e6/src/train/prom/prom_query.py#L26. 
Could I left someone work on that later if not match?

Reply:
Yes, I am happy to have this validated later. 
Could you please add a todo .. something like  ` // TODO: ensure trailing , is accepted`